src/Aspire.Hosting.Azure/Provisioning/Internal/BicepCompiler.cs

```diff
@@ -0,0 +1,79 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Text;
+using Aspire.Hosting.Dcp.Process;
+
+namespace Aspire.Hosting.Azure.Provisioning.Internal;
+
+/// <summary>
+/// Default implementation of <see cref="IBicepCompiler"/>.
+/// </summary>
+internal sealed class BicepCliCompiler : IBicepCompiler
+{
+    public async Task<string> CompileBicepToArmAsync(string bicepFilePath, CancellationToken cancellationToken = default)
+    {
+        var azPath = FindFullPathFromPath("az") ?? throw new AzureCliNotOnPathException();
```

Comment:
Update this look-up to use the more performant `bicep` command first, then `az` then throw if neither are found. The arguments should be updated depending on whether the `az` or `bicep` command was found.